app.ts

```diff
@@ -30,110 +22,123 @@ class Grid {
                 this.headers = headers;
                 this.pageInit();
 
-                //adds initial records to row objects to be added to grid on page load.
-                $.getJSON(host + "/records?from=" + this.from + "&to=" + this.to, (data) => { })
-                    .done((records) => {
-                        this.drawGrid(records);
-                    });
+                // adds initial records to row objects to be added to grid on page load.
+                this.drawBasicFromToGrid();
             });
     }
 
-    //(re)initializes the page deafault values
-    pageInit() {
+    // (re)initializes the page deafault values
+    private pageInit(): void {
         $('#grid').empty();
-        this.$row.empty();
-        this.$column.empty();
-
-        //add headers to row object to be added to the grid on initialization.
-        for (let col of this.headers) {
-            this.$row.append(this.$header.clone().text(col));
-        }
 
-        $("#grid").append(this.$row.clone());
+        // add headers to row object to be added to the grid on initialization.
+        this.drawRow("header", this.headers);
     }
 
-    //iterates through records and adds them as rows to the grid
-    drawGrid(records: string[][]) {
+    // iterates through records and adds them as rows to the grid
+    private drawGrid(records: string[][]): void {
         for (let record of records) {
-            this.$row.empty();
+            this.drawRow("column", record);
+        }
+    }
 
-            for (let col of record) {
-                this.$row.append(this.$column.clone().text(col));
-            }
+    // draws individual rows by appending column objects to it and appending the row object to the grid.
+    private drawRow(css_class: string, record: string[]): void {
+        let $colu = $("<div />", {
+            class: css_class
+        });
+
+        let row = this.newRow();
 
-            $("#grid").append(this.$row.clone());
+        for (let col of record) {
+            row.append($colu.clone().text(col));
         }
+
+        $("#grid").append(row.clone());
```

Comment:
You shouldn't need to clone() the row here, just add the row that's been newly created. Cloning it makes the operation unnecessarily expensive because you're essentially creating two rows when you only need one.